webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx

```diff
@@ -10,15 +10,13 @@ type Props = {
     intl: IntlShape;
 }
 
-class WarningIcon extends React.PureComponent<Props> {
-    public render(): JSX.Element {
-        return (
-            <i
-                className={classNames('fa fa-warning', this.props.additionalClassName)}
-                title={this.props.intl.formatMessage({id: 'generic_icons.warning', defaultMessage: 'Warning Icon'})}
-            />
-        );
-    }
-}
+const WarningIcon = ({additionalClassName, intl}: Props) => {
+    return (
+        <i
+            className={classNames('fa fa-warning', additionalClassName)}
+            title={intl.formatMessage({id: 'generic_icons.warning', defaultMessage: 'Warning Icon'})}
+        />
+    );
+};
 
-export default injectIntl(WarningIcon);
+export default React.memo(injectIntl(WarningIcon));
```

Comment:
For function components we don`t need `injectIntl`. We can directly use the hook `useIntl`. Can you do the following?
- Remove injectIntl from this line
- Use the `useIntl` hook inside the component
- Remove intl from the props
- Update the snapshots (they are going to break :sweat_smile: )

Reply:
@larkox, I have implemented the changes. 😄